helix-core/src/main/java/org/apache/helix/task/TaskUtil.java

```diff
@@ -201,6 +202,9 @@ protected static HelixConfigScope getResourceConfigScope(String clusterName, Str
    */
   protected static JobContext getJobContext(HelixPropertyStore<ZNRecord> propertyStore,
       String jobResource) {
+    if (jobResource == null) {
+      throw new InvalidParameterException("job name is null");
```

Comment:
NIT: Let's make the sentence more complete, something like : "Null job name is now allowed."

Reply:
TFTR. Updated.